rs/replicated_state/src/metadata_state.rs

```diff
@@ -1557,10 +1558,12 @@ impl Streams {
     /// subnet.
     pub fn push(&mut self, destination: SubnetId, msg: RequestOrResponse) {
         if let RequestOrResponse::Response(response) = &msg {
-            *self
-                .responses_size_bytes
-                .entry(response.respondent)
-                .or_default() += msg.count_bytes();
+            if response.deadline == NO_DEADLINE {
+                *self
+                    .guaranteed_responses_size_bytes
+                    .entry(response.respondent)
+                    .or_default() += msg.count_bytes();
+            }
```

Comment:
Why are we tracking this here and also below in the `StreamHandle` isn't this somehow duplicate?

Reply:
No, there are two ways to push messages into a stream: via `Streams::push(destination, mst)` or via `StreamHandle::push(msg)`.

`StreamHandle` is just a way of providing a client with something like a `Stream` that in the background also updates `Streams::guaranteed_responses_size_bytes` (which is why you can only have one `StreamHandle` at a time). Else, we would have to recompute `guaranteed_responses_size_bytes` whenever the `Stream` was replaced into `Streams`.